lib/src/data/article/ny_times_paginated_rest_repository.dart

```diff
@@ -0,0 +1,36 @@
+import 'package:allthenews/src/domain/common/page.dart';
+import 'package:allthenews/src/domain/model/article.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_cached_repository.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_paginated_repository.dart';
+import 'package:allthenews/src/domain/nytimes/ny_times_repository.dart';
+import 'package:allthenews/src/domain/settings/popular_news_criterion.dart';
+import 'package:allthenews/src/domain/settings/settings_repository.dart';
+
+class NyTimesPaginatedRestRepository extends NYTimesPaginatedRepository {
+  final NYTimesRepository _nyTimesRepository;
+  final NyTimesCachedRepository _nyTimesCachedRepository;
+  final SettingsRepository _settingsRepository;
+
+  NyTimesPaginatedRestRepository(
+    this._nyTimesRepository,
+    this._nyTimesCachedRepository,
+    this._settingsRepository,
+  );
+
+  @override
+  Future<List<Article>> getMostPopularArticlesPage(Page page) async {
+    final PopularNewsCriterion popularNewsCriterion = await _settingsRepository.getPopularNewsCriterion();
+    if (page.number == 0) {
+      await _nyTimesRepository.getMostPopularArticles().then((articles) => _nyTimesCachedRepository.saveMostPopularArticles(articles, popularNewsCriterion));
```

Comment:
to się nigdy nie wykona bo w notifierze ustawiasz na sztywno pierwszą wartość na 1